catalog/registry/catalog-registry-federationadminserviceimpl/src/test/java/ddf/catalog/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -294,10 +336,285 @@ public void testGetRegistryMetacardsUnsupportedQueryException() throws Exception
+        List<Metacard> metacards = new ArrayList<>();
+        Metacard findThisMetacard = getTestMetacard();
+        findThisMetacard.setAttribute(new AttributeImpl(Metacard.METADATA, "ThisIsMyMetadata"));
+        metacards.add(findThisMetacard);
+
+        when(parser.unmarshal(any(ParserConfigurator.class),
+                eq(JAXBElement.class),
+                any(InputStream.class))).thenThrow(ParserException.class);
+        doReturn(metacards).when(fasi)
+                .getRegistryMetacards();
+
+        List<RegistryPackageType> registryPackages = fasi.getLocalRegistryObjects();
+
+        verify(fasi).getRegistryMetacards();
+        verify(parser, times(1)).unmarshal(any(ParserConfigurator.class),
+                eq(JAXBElement.class),
+                any(InputStream.class));
+
+        assertThat(registryPackages, empty());
+    }
+
+    @Test
+    public void testUpdateLocalEntry() throws Exception {
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(Metacard.ID, "SomeMetacardId"));
+
+        fasi.updateLocalEntry(metacard);
+
+        verify(catalogFramework).update(any(UpdateRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testUpdateLocalEntryWithNoId() throws Exception {
+        Metacard metacard = getTestMetacard();
+
+        fasi.updateLocalEntry(metacard);
+
+        verify(catalogFramework, never()).update(any(UpdateRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testUpdateLocalEntryWithSourceUnavailableException() throws Exception {
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(Metacard.ID, "SomeMetacardId"));
+
+        when(catalogFramework.update(any(UpdateRequest.class))).thenThrow(SourceUnavailableException.class);
+
+        fasi.updateLocalEntry(metacard);
+
+        verify(catalogFramework).update(any(UpdateRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testUpdateLocalEntryWithIngestException() throws Exception {
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(Metacard.ID, "SomeMetacardId"));
+
+        when(catalogFramework.update(any(UpdateRequest.class))).thenThrow(IngestException.class);
+        fasi.updateLocalEntry(metacard);
+
+        verify(catalogFramework).update(any(UpdateRequest.class));
+    }
+
+    @Test
+    public void testUpdateLocalEntryRegistryPackage() throws Exception {
+        String registryId = "someRegistryId";
+        List<String> ids = new ArrayList<>();
+        ids.add(registryId);
+
+        RegistryPackageType registryPackage = getTestRegistryPackage();
+        registryPackage.setId(registryId);
+
+        Metacard metacard = getTestMetacard();
+
+        Metacard existingMetacard = getTestMetacard();
+        existingMetacard.setAttribute(new AttributeImpl(Metacard.ID, registryId));
```

Comment:
This should be Metacard.ID.
Created a metacardId string and use that instead of registryId. To avoid confusion.